core/templates/services/favicon.service.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for changing favicon dynamically.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FaviconService {
+  constructor() {}
+
+  setFavicon(faviconUrl: string): void {
+    const link: HTMLLinkElement = (
+      document.querySelector('link[rel*="icon"]') ||
```

Comment:
I think accessing `document` directly is not preferred, you might need something like windowRef.nativeWindow.document.

/cc @srijanreddy98 @ashutoshc8101 to confirm

Reply:
I have changed this to windowRef.nativeWindow.document for now and waiting for other reviewers' opinion.